Chapter07/pretty_print_on_the_fly.cpp

```diff
@@ -4,12 +4,15 @@
 using namespace std;
 
 class format_guard {
-    decltype(cout.flags()) f {cout.flags()};
+    ostream& _os;
+    ios::fmtflags _f;
```

Comment:
Yes, i said "globally", sorry for my clumsy english. In different projects i worked at it was considered better practice to use suffix and not prefix because this way no one slides into the habit of by accident doing the same in the global namespace one day.